### DISCUSSION TITLE
Question about the `OR` operator

### DISCUSSION THREAD

**Discussion by tomodachi94** [retorquere/zotero-better-bibtex#2530] [Q&A] [answered]
Hey, I'm having some trouble with Zotero and BetterBibTeX.

Scenario: I have several items in my Zotero library that do not have any specific author; they only have a publisher.

I want to have my BibTeX citekey be set to one of these two, depending on if the author field is present:

```
johnDoe.2023.zoteroForumPost
johnDoePublishing.2023.someVideo
```

Here's what I have so far:

```
auth.lower|Publisher.lower + "." + year + "." + shorttitle(3,3)
```

However, if the author field is present, the citekey gets set to this:

```
johnDoe
```

The items that have a publisher with no authors have the expected output.

I've also tried this:

```
auth.lower + "." + year + "." + shorttitle(3,3)|Publisher.lower + "." + year + "." + shorttitle(3,3)
```

This doesn't work because the `OR` operator expects 'null', but instead gets the string `.2021.MyFavoriteArticle`.

**Comment by retorquere** [comment]
I'll need a debug log for this item to be able to tell precisely.

**Reply by retorquere** [reply, marked as answer]
But maybe

```
auth.lower.len + "." + year + "." + shorttitle(3,3)|Publisher.lower + "." + year + "." + shorttitle(3,3)
```

already does what you want?

**Reply by tomodachi94** [reply]
@retorquere this does what I need it to! Thank you so much!
